tests/e2e/test_pgsql_node_drain.py

```diff
@@ -0,0 +1,139 @@
+"""
+Module to perform PGSQL workload
+"""
+import logging
+import pytest
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import templating, utils
+from ocs_ci.utility.utils import run_cmd, TimeoutSampler
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import node
+from tests.sanity_helpers import Sanity
+from ocs_ci.framework.testlib import (
+    E2ETest, google_api_required, workloads, ignore_leftovers
+)
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.utility.spreadsheet.spreadsheet_api import GoogleSpreadSheetAPI
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='function')
+def ripsaw(request):
+
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+
+    def teardown():
+        ripsaw.cleanup()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+@ignore_leftovers
+@workloads
+@google_api_required
+@pytest.mark.polarion_id("OCS-806")
+class TestPgSQLNodeReboot(E2ETest):
+    """
+    Test running PGSQL and with Ceph pods respin
+    """
+    @pytest.fixture()
+    def pgsql_setup(self, ripsaw):
+        """
+        PGSQL test setup
+        """
+        # Deployment ripsaw and postgres database
+        log.info("Deploying postgres database")
+        ripsaw.apply_crd(
+            'resources/crds/'
+            'ripsaw_v1alpha1_ripsaw_crd.yaml'
+        )
+        ripsaw.setup_postgresql()
+
+        # Initialize Sanity instance
+        self.sanity_helpers = Sanity()
+
+    @pytest.mark.usefixtures(pgsql_setup.__name__)
+    def test_run_pgsql_node_drain(self, transactions=900, node_type='master'):
```

Comment:
Do you also plan to have other node types covered? Maybe this can be parameterized. Draining of master node may only have limited impact to ceph pods as none of the ceph pods spin up on master node in the current implementation of ocs.

Reply:
Planning for other node type to cover in a separate PR.